### DISCUSSION TITLE
Command Pallette, output date string

### DISCUSSION THREAD

**Discussion by deterset** [microsoft/terminal#17470] [Ideas]
Can we program a key (like F7 for example) to output  a formatted date string to the terminal.
We are using that function in Mocha Telnet but I would like to replace it with this awesome terminal program.

Maybe it is possible and I just don't know how to do it.

**Comment by deterset** [comment]
Just another thought. I suppose I could write a script to run once a day and update the json config file and re-program a key each day to output a sequence of characters like "06/25/2024"  for today. Anyone else have any thoughts ?

**Reply by DHowett** [reply]
You know, it would be really cool if we had a scripting language for input actions. It's not something we're likely to add, but maybe an extension could.

As for your other idea: yeah! [Fragment Extensions](https://learn.microsoft.com/en-us/windows/terminal/json-fragment-extensions) can add actions, like `sendInput`, with custom content. Once 1.22 is out, you will be able to set a key binding to one of those actions as well (sorry, this is not all done and released to the public yet!)

That might go a long way to making this possible. If you rewrite the fragment JSON with a new date in it, and somehow trigger Terminal to reload its settings, your key binding would always work to insert the current date. 🙂
